### ISSUE TITLE
Problem in reproducing the paper results

### ISSUE THREAD

**Issue by PKULiuHui** [lancopku/livebot#1]
Hello, I have read your paper and trained the transformer model with the processed dataset. However, I can't reproduce the results in the paper.

The results in the paper are showed below:
![pic1](https://user-images.githubusercontent.com/32560313/62820366-ed768700-bb95-11e9-9814-419882480573.png)
We can see that the model trained with both video and comment performs best.

I download the code and the processed dataset, train the model on a single GPU(it takes about 2.5 days for 50 epoches), and test the model on the final epoch, just like the instructions in README.md(I also test other checkpoints and the results are close). I set n_img to 0 and n_com to 0 respectively and train two another models to get the Comment_Only and Video_Only results.
![pic2](https://user-images.githubusercontent.com/32560313/62820462-31b65700-bb97-11e9-9982-665ac020601d.png)
The values are much higher than the results in your paper and Video_Only model shows an extremely high performance, which is counter-intuitive.

So I check the source code and find something strange(maybe a bug?). The line 155 in transformer.py is showed below. 
<img width="376" alt="pic4" src="https://user-images.githubusercontent.com/32560313/62820561-6e368280-bb98-11e9-9281-2ff666fef9bf.png">
It returns the loss rank of 100 candidate comments for each test item and the evaluation metrics are calculated accordingly. It should be a list ranked by the candidates' log-likelihood scores descendingly. However, the CrossEntropy Loss is contrary to the log-likelihood score, a lower loss means a higher log-likelihood score. So I think the loss should be ranked ascendingly. I fix the code line and run the test again.
![pic3](https://user-images.githubusercontent.com/32560313/62820626-97a3de00-bb99-11e9-8a03-c7aca6acb43a.png)
This time, the model trained with both video and comment performs best as wished. However, the value is much lower than the results in your paper.

So I have some questions and expect your reply:
1. Is there really a bug in transformer.py line 155 or it is just my misunderstanding?
2. Why the results are so different from the paper(too high and strange on the original code, too low on my fixed code)? Is the processed dataset is what you use in the paper? Or is there something else wrong in the code?
3. What's your result after runing the code in the repository? I have trained two times and get close results showed above.

**Comment by fireflyHunter** [comment]
I am getting similar scores, any update on this?
